spec/intro.dd

```diff
@@ -8,59 +8,51 @@ $(SPEC_S Introduction,
-syntax directed editors relatively easy to produce. It also is possible to
-compress D source by storing it in $(SINGLEQUOTE tokenized) form.)
+$(H3 Environments)
 
-$(OL
-        $(LI $(B source character set)$(BR)
+$(P The data-processing-system in which a D program is translated from source to executable format
+will be called the translation environment and the data-processing-system in which the translated
+D program is executed will be called the execution environment. The translation environment and
+the execution environment need not have the same system architecture.)
 
-        The source file is checked to see what character set it is,
-        and the appropriate scanner is loaded. ASCII and UTF
-        formats are accepted.
-        )
+$(H3 Program Structure)
 
-        $(LI $(B script line) $(BR)
+$(P A D program is kept in units called source files. Each source file contains a unit of translation
+called the module. D modules need not all be translated at the same time. Previously translated
+D modules may be individually preserved or in libraries. The separate modules of a program communicate
+by calls to functions whose identifers have external linkage, manipulation of objects whose identifers
+have external linkage, or manipulation of data files. Modules may be separately translated and then
+later linked to produce an executable program.)
 
-        If the first line starts with "$(HASH)!", then that line
-        is ignored.
-        )
+$(H3 Translation Phases)
 
-        $(LI $(B lexical analysis)$(BR)
+$(P D programs are translated in multiple phases.)
+
+$(OL
 
-        The source file is divided up into a sequence of tokens.
-        $(DDSUBLINK spec/lex, specialtokens, Special tokens)
-        are replaced with other tokens.
-        $(GLINK_LEX SpecialTokenSequence)s
-        are processed and removed.
+        $(LI $(B Lexical analysis)$(BR)
+
+        The source file is translated into a sequence of tokens.
         )
 
-        $(LI $(B syntax analysis)$(BR)
+        $(LI $(B Syntax analysis)$(BR)
 
         The sequence of tokens is parsed to form syntax trees.
         )
 
-        $(LI $(B semantic analysis)$(BR)
+        $(LI $(B Semantic analysis)$(BR)
 
         The syntax trees are traversed to declare variables, load symbol tables, assign
         types, and in general determine the meaning of the program.
         )
 
-        $(LI $(B optimization)$(BR)
-
-        Optimization is an optional pass that tries to rewrite the program
-        in a semantically equivalent, but faster executing, version.
-        )
-
-        $(LI $(B code generation)$(BR)
+        $(LI $(B Code generation)$(BR)
 
-        Instructions are selected from the target architecture to implement
-        the semantics of the program. The typical result will be
-        an object file, suitable for input to a linker.
+        Following optional optimizations, instructions targetted for the execution
```

Comment:
Remove `optimal`.

Reply:
Hi, did you mean `optional`? I put that in as presumably optimization is an optional step. The original text stated `Optimization is an optional pass`.

Reply:
Whoops, yes,